homeassistant/components/owlet/config_flow.py

```diff
@@ -0,0 +1,188 @@
+                            CONF_OWLET_EXPIRY: token[CONF_OWLET_EXPIRY],
+                            CONF_OWLET_REFRESH: token[CONF_OWLET_REFRESH],
+                        },
+                        options={CONF_SCAN_INTERVAL: POLLING_INTERVAL},
+                    )
+                except OwletDevicesError:
+                    errors["base"] = "no_devices"
+
+            except OwletEmailError:
+                errors["base"] = "invalid_email"
+            except OwletPasswordError:
+                errors["base"] = "invalid_password"
+            except OwletCredentialsError:
+                errors["base"] = "invalid_credentials"
+            except Exception:  # pylint: disable=broad-except
+                _LOGGER.exception("Unexpected exception")
+                errors["base"] = "unknown"
+
+        return self.async_show_form(
+            step_id="user", data_schema=STEP_USER_DATA_SCHEMA, errors=errors
+        )
+
+    @staticmethod
+    @callback
+    def async_get_options_flow(config_entry: ConfigEntry) -> OptionsFlowHandler:
+        """Get the options flow for this handler."""
+        return OptionsFlowHandler(config_entry)
+
+    async def async_step_reauth(self, user_input: Mapping[str, Any]) -> FlowResult:
+        """Handle reauth."""
+        self.reauth_entry = self.hass.config_entries.async_get_entry(
+            self.context["entry_id"]
+        )
+        return await self.async_step_reauth_confirm()
+
+    async def async_step_reauth_confirm(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Dialog that informs the user that reauth is required."""
+        assert self.reauth_entry is not None
+        errors: dict[str, str] = {}
+
+        if user_input is not None:
+            entry_data = self.reauth_entry.data
+            owlet_api = OwletAPI(
+                entry_data[CONF_REGION],
+                entry_data[CONF_USERNAME],
+                user_input[CONF_PASSWORD],
+                session=async_get_clientsession(self.hass),
+            )
+            try:
+                token = await owlet_api.authenticate()
+                if token:
+                    self.hass.config_entries.async_update_entry(
+                        self.reauth_entry, data={**entry_data, **token}
+                    )
+
+                await self.hass.config_entries.async_reload(self.reauth_entry.entry_id)
+
+                return self.async_abort(reason="reauth_successful")
+
+            except OwletPasswordError:
+                errors["base"] = "invalid_password"
+            except Exception:  # pylint: disable=broad-except
+                _LOGGER.exception("Error reauthenticating")
+
+        return self.async_show_form(
+            step_id="reauth_confirm",
+            data_schema=vol.Schema({vol.Required(CONF_PASSWORD): str}),
+            errors=errors,
+        )
+
+
+class OptionsFlowHandler(config_entries.OptionsFlow):
```

Comment:
Personally to keep this PR small, I'd remove the Options Flow for now. Just another chunk of code that a core maintainer will have to review. the smaller the PR the more likely it is for this to get merged sooner